desktop/core/src/desktop/lib/importer/operations.py

```diff
@@ -0,0 +1,685 @@
+      - total_rows: Approximate total row count
+      - delimiters: Information about detected delimiters
+
+  Raises:
+    Exception: If there's an error processing the delimited file
+  """
+  try:
+    # Auto-detect header if not specified
+    if has_header is None:
+      fh.seek(0)
+      try:
+        # Read 16 KiB sample
+        sample = fh.read(16 * 1024).decode('utf-8', errors='replace')
+
+        has_header = csv.Sniffer().has_header(sample)
+        LOG.info(f"Auto-detected header for delimited file: {has_header}")
+      except Exception as e:
+        LOG.warning(f"Error auto-detecting header in delimited file: {e}")
+        # Default based on file type - most CSVs have headers
+        has_header = True if file_type in ['csv', 'tsv'] else False
+    else:
+      has_header = coerce_bool(has_header)
+
+    # Reset file position and read file content
+    fh.seek(0)
+
+    df = pl.read_csv(
+      BytesIO(fh.read()),
+      separator=field_separator,
+      quote_char=quote_char,
+      eol_char='\n' if record_separator == '\r\n' else record_separator,
+      has_header=has_header,
+      ignore_errors=True,
+    )
+
+    # Check if the dataframe is empty
+    if df.height == 0:
+      return {'type': file_type, 'has_header': has_header, 'columns': [], 'preview_data': {}, 'total_rows': 0}
+
+    # Get schema and preview data
+    schema = df.schema
+
+    # Limit preview to specified number of rows for performance
+    preview_data = df.head(preview_rows).to_dict(as_series=False)
+
+    # Create detailed column metadata with SQL type mapping
+    columns = []
+    for i, col in enumerate(df.columns):
+      col_type = str(schema[col])
+      sql_type = map_polars_to_sql(dialect, col_type)
+
+      columns.append(
+        {
+          'name': col,
+          'type': sql_type,
+          'index': i,  # Add index for ordering
+        }
+      )
+
+    result = {'type': file_type, 'has_header': has_header, 'columns': columns, 'preview_data': preview_data, 'total_rows': df.height}
+
+    return result
+
+  except Exception as e:
+    message = f"Error previewing delimited file: {e}"
+    LOG.error(message, exc_info=True)
+    raise Exception(message)
+
+
+def map_polars_to_sql(dialect: str, polars_type: str) -> str:
+  """
+  Map a Polars dtype to the corresponding SQL type for a given dialect.
+
+  Supports all Polars dtypes as listed in the Polars docs:
+  Int8, Int16, Int32, Int64, UInt8, UInt16, UInt32, UInt64,
+  Float32, Float64, Decimal, Boolean, Utf8/String, Categorical, Enum,
+  Binary, Date, Time, Datetime, Duration, Array, List, Struct,
+  Object, Null, Unknown.
+
+  Args:
+    dialect: One of "hive", "impala", "trino", "phoenix", "sparksql".
+    polars_type: Polars dtype name as string.
+
+  Returns:
+    A string representing the SQL type.
+
+  Raises:
+    ValueError: If the dialect or polars_type is not supported.
+  """
+  # Base mapping for most engines (Hive, Impala, SparkSQL)
+  base_map = {
```

Comment:
Can we keep these mappings to somewhere outside of the function? It is a static mapping and probably doesn't need to be recreated on every call.

Reply:
Nice suggestion, I have incorporated the changes.

Also I've created a new API endpoint to fetch this mapping so that UI can use it for populating the dropdown for users to choose the column types if they want to change that.